lib/saluki-config/src/dynamic/diff.rs

```diff
@@ -0,0 +1,47 @@
+//! Functions for diffing configuration values.
+
+use super::event::ConfigChangeEvent;
+
+/// Diffs two configuration values and returns a list of changes.
+pub fn diff_config(old_config: &figment::value::Value, new_config: &figment::value::Value) -> Vec<ConfigChangeEvent> {
```

Comment:
I think we just want a basic unit test, or two, for this. Primarily to show that we emit the right event type for modifications vs additions, and also to show that for dictionaries, we emit an event _per_ key rather than just saying "yeah, these two dicts are different".